### PR TITLE
Change Image.interpolation to Enum

### LINKED ISSUE #132: Remove buggy interpolation modes
## 🐛 Bug

Some of Vispy interpolation modes for images are very buggy and lead to unsightly
visual artefacts such as black or white vertical lines.
Here is a list of buggy interpolation modes:

- quadric (white lines, sometimes fails entirely)
- sinc (black lines)

We should remove any buggy interpolation scheme,
we are more interpolation than is healthy to have.

## To Reproduce

1. run 'add_images.py'
2. change interpolation schemes and check eac one.

image

For quadric, the white lines are faint but there:
image

## Expected behavior

no weird lines.

## Environment

 - napari-gui Version (e.g., 1.0): master
 - OS (e.g., Linux): OSX
 - Python version: 3.7
 - Any other relevant information: None

## Additional context

None

### PR DESCRIPTION
**Pull request by kevinyamauchi**
# Description
As discussed in #265, this PR introduces an Enum to represent the Image layer interpolation options (`Interpolation`). The `Interpolation` is inherits from the `StringEnum` class introduced in  #311 and thus has string values and is implemented such that the `Layer.blending` property can be set with a string and returns a string. Further, we have removed the ability to set the buggy quadric and sinc interpolation modes (per #132).

Previously, `Image.interpolation` was stored as an index (`Image.interpolation_index`) and the `Image.interpolation` property and setter methods converted between the index and a string name via some helper functions in `/napari/util/interpolation.py'. I checked the code base and ran the examples listed below and it appeared that nothing else referenced that file, so I deleted it. Let me know if that was a mistake!

Similarly, `Image.interpolation_names` (tuples of interpolation mode names) appeared unused so I deleted it. However, if people think it's important to have available, I am happy to add it back in.

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

# Reference
This is PR#3 in #265 and closes #132.

# How has this been tested?
To test compatibility with layers, I ran the following examples
- [x] `Layer.Shapes`: `/examples/add_shapes.py`
- [x] `Layer.Markers`: `/examples/add_markers.py`
- [x] `Layer.Labels`: `/examples/labels-2d.py`
- [x] `Layer.Vectors`: `/examples/add_vectors_coords.py`
- [x] `Layer.Pyramid`: `/examples/add_pyramids.py`

To test the ability to set `Image.interpolation` with a string, I ran:
- [x] `/examples/add_image.py`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by sofroniewn — CHANGES_REQUESTED** [review]
This looks great! Can we also address #132 right now too and drop support for `quadric` and `sinc` modes due to the bugs described in that issue.

File: napari/layers/_image_layer/_constants.py
Comment (sofroniewn) [inline napari/layers/_image_layer/_constants.py]: drop quadric and sinc as described in #132

Comment (kevinyamauchi) [reply napari/layers/_image_layer/_constants.py]: Great point. Done!

**Comment by kevinyamauchi** [comment]
@sofroniewn thanks for the comments. I dropped the quadric and sinc interpolation modes as you suggested. I am happy to wait to merge until #322 (`black` PR) is landed.

File: napari/layers/_image_layer/model.py
Comment (kne42) [inline napari/layers/_image_layer/model.py]: I would consider this "public" so this should be the string version:
```suggestion
    default_interpolation = str(Interpolation.NEAREST)
```

File: napari/layers/_image_layer/model.py
Comment (kne42) [inline napari/layers/_image_layer/model.py:314]: IMO this should just be a wrapper around `self._node.interpolation` which calls into question if we really need an Enum at all.

File: napari/layers/_image_layer/model.py
Comment (kne42) [inline napari/layers/_image_layer/model.py]: this docstring should include all possible methods listed in set format as the type: `{'nearest', 'bicubic', ...}`

File: napari/layers/_image_layer/_constants.py
Comment (kne42) [inline napari/layers/_image_layer/_constants.py:6]: do you think this Enum needs its own file?

Comment (sofroniewn) [reply napari/layers/_image_layer/_constants.py:6]: I like the constants having their own files for consistencies sake, and the fact that we may add more constants here too. Maybe some of the colormap processing stuff should be in here too instead of the model file. Thoughts @jni?

File: napari/layers/_image_layer/model.py
Comment (sofroniewn) [reply napari/layers/_image_layer/model.py:314]: I like that the Enum allows us more control over which modes we expose and don't (i.e. it lets us drop `sinc`)

File: napari/layers/_image_layer/model.py
Comment (kevinyamauchi) [reply napari/layers/_image_layer/model.py]: Good point. Fixed!

File: napari/layers/_image_layer/model.py
Comment (kevinyamauchi) [reply napari/layers/_image_layer/model.py]: Fixed. Thanks!

File: napari/layers/_image_layer/_constants.py
Comment (kevinyamauchi) [reply napari/layers/_image_layer/_constants.py:6]: I agree that we should be consistent across the layer types. Given that the other layers store things like Enum's in a `_constants.py` file, I think we should do the same here. As @sofroniewn mentioned, we may add other things here in the future anyway.

**Comment by kevinyamauchi** [comment]
Based on our conversation at the sync meeting last Thursday, I think we decided to move forward with the `Interpolation` Enum and that it should live in the `_constants.py` file. I think I have addressed the remaining comments. Please let me know if there's anything else you would like me to change!

**Review by kne42 — APPROVED** [review]

**Comment by sofroniewn** [comment]
I have update this with the `black` formatting and will merge once all the tests pass

**Review by sofroniewn — APPROVED** [review]
LGTM will merge

**Pull request merged**
